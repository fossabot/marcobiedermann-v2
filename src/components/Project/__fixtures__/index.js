const project = {
  id: '6c5a5baa-8c8b-5580-8061-b1cad23712cc',
  slug: 'demo-project',
  title: 'Demo Project',
};

export default project;
