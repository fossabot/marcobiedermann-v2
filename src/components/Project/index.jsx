import { Link } from 'gatsby';
import PropTypes from 'prop-types';
import React from 'react';

const Project = props => {
  const { title, slug } = props;

  return (
    <div>
      <h2>
        <Link to={`/projects/${slug}`}>{title}</Link>
      </h2>
    </div>
  );
};

Project.propTypes = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
};

export default Project;
