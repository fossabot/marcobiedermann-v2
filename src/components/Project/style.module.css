.project__caption {
  align-items: center;
  background-color: rgba(0, 0, 0, 0.9);
  bottom: 0;
  color: #fafafa;
  display: flex;
  justify-content: center;
  left: 0;
  opacity: 0;
  position: absolute;
  right: 0;
  text-align: center;
  top: 0;
  transition: opacity 0.5s;
}

.project__figure {
  position: relative;
}

.project__link:focus .project__caption,
.project__link:hover .project__caption {
  opacity: 1;
}
