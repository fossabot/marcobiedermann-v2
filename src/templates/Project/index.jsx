import { graphql } from 'gatsby';
import PropTypes from 'prop-types';
import React from 'react';
import Section from '../../components/Section';

const ProjectTemplate = props => {
  const {
    data: {
      contentfulProject: { agency, client, role, title },
    },
  } = props;

  return (
    <div>
      <Section>
        <h1>{title}</h1>
        <ul>
          <li>
            <strong>Client:</strong> {client}
          </li>
          <li>
            <strong>Agency:</strong> {agency}
          </li>
          <li>
            <strong>Role:</strong> {role}
          </li>
        </ul>
      </Section>
    </div>
  );
};

ProjectTemplate.propTypes = {
  data: PropTypes.shape({
    contentfulProject: PropTypes.shape({
      agency: PropTypes.string,
      client: PropTypes.string,
      role: PropTypes.string,
      title: PropTypes.string,
    }),
  }),
};

ProjectTemplate.defaultProps = {
  data: null,
};

export const pageQuery = graphql`
  query ProjectBySlug($slug: String!) {
    contentfulProject(slug: { eq: $slug }) {
      agency
      client
      id
      role
      slug
      tags
      title
      url
    }
  }
`;

export default ProjectTemplate;
