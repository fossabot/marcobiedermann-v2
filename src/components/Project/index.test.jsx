import React from 'react';
import renderer from 'react-test-renderer';
import Project from '.';

describe('Project component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Project title="Project Title" slug="project-title" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
