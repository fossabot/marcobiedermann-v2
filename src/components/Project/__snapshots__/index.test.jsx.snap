// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project component should render correctly 1`] = `
<div>
  <h2>
    <a
      href="/projects/project-title"
    >
      Project Title
    </a>
  </h2>
</div>
`;
